src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -556,6 +571,42 @@ private void createMesosTask(Offer offer, Request request) {
 
     results.put(taskId, new Result(request.result, new Mesos.JenkinsSlave(offer.getSlaveId()
         .getValue())));
+    finishedTasks.add(taskId);
+  }
+
+  /**
+   * Checks if the given taskId already exists or just finished running. If it has, then refuse the offer.
+   * @param offer The offer to check if the task is already running
+   * @param taskId The task id
+   * @return True if the task already exists, false otherwise
+   */
+  @VisibleForTesting
```

Comment:
Yep, can't add them until after a couple other pull requests are merged in though. I'll follow up with then when I get done time

On June 21, 2015 9:51:15 PM CDT, Vinod Kone <notifications@github.com> wrote:
>> @@ -556,6 +571,42 @@ private void createMesosTask(Offer offer,
>Request request) {
>>  
>>      results.put(taskId, new Result(request.result, new
>Mesos.JenkinsSlave(offer.getSlaveId()
>>          .getValue())));
>> +    finishedTasks.add(taskId);
>> +  }
>> +
>> +  /**
>> +   * Checks if the given taskId already exists or just finished
>running. If it has, then refuse the offer.
>> +   * @param offer The offer to check if the task is already running
>> +   * @param taskId The task id
>> +   * @return True if the task already exists, false otherwise
>> +   */
>> +  @VisibleForTesting
>
>why this annotation here and below? are you planning to add tests? it
>would be great, if you plan to by the way!
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/jenkinsci/mesos-plugin/pull/124/files#r32902634

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.